src/ol/control/Attribution.js

```diff
@@ -192,10 +200,16 @@ class Attribution extends Control {
    */
   collectSourceAttributions_(frameState) {
     const layers = this.getMap().getAllLayers();
-    const visibleAttributions = Array.from(
+    let visibleAttributions = Array.from(
       new Set(layers.flatMap((layer) => layer.getAttributions(frameState))),
     );
 
+    if (this.staticAttribution_ !== undefined) {
```

Comment:
I'm not entirely happy with how this works (recreating the array again). I feel there must be a way to do it in one go, but simply pushing to the `visibleAttributions` array didn't work as the de-duplication of using a `Set` is then lost when adding the static attribution. Suggestions welcome